_studio/shared/src/libmfx_allocator.cpp

```diff
@@ -41,7 +44,11 @@ mfxStatus mfxDefaultAllocator::AllocBuffer(mfxHDL pthis, mfxU32 nbytes, mfxU16 t
     if(!mid)
         return MFX_ERR_NULL_PTR;
     mfxU32 header_size = ALIGN32(sizeof(BufferStruct));
-    mfxU8 *buffer_ptr=(mfxU8 *)malloc(header_size + nbytes + DEFAULT_ALIGNMENT_SIZE);
+#if defined (MFX_VA_LINUX)
+    mfxU8 *buffer_ptr = (mfxU8 *)aligned_alloc(sysconf(_SC_PAGESIZE), header_size + nbytes + DEFAULT_ALIGNMENT_SIZE);
```

Comment:
Need to zero memory to align with Windows

Reply:
Returned malloc for non-linux case.

Reply:
why don't we use ```std::align``` instead? I believe best way would be to refactor this class in base of ```std::aligned_storage```